### PR TITLE
fix: add cache context

### LINKED ISSUE #11: Add a cache context

### LINKED ISSUE #15: Set autocomplete with override as default widget

### PR DESCRIPTION
**Pull request by chrfritsch**

### CODE REVIEW

**Comment by chrfritsch** [comment]
This fixes #11

File: src/Plugin/Field/FieldType/EntityReferenceOverrideItem.php [hunk 1]
```diff
@@ -56,16 +56,18 @@ public static function schema(FieldStorageDefinitionInterface $field_definition)
    * {@inheritdoc}
    */
   public function __get($name) {
-    if ($name == 'entity' && !empty($this->values['overwritten_property_map'])) {
```
Comment (ol0lll) [inline src/Plugin/Field/FieldType/EntityReferenceOverrideItem.php:59]: entity_reference_override_property_map

File: entity_reference_override.module [hunk 2]
```diff
@@ -12,7 +12,7 @@ use Drupal\entity_reference_override\Exception\ReadOnlyEntityException;
  * Implements hook_entity_presave().
  */
 function entity_reference_override_entity_presave(EntityInterface $entity) {
-  if (isset($entity->overwritten) && $entity->overwritten === TRUE) {
+  if (isset($entity->overwritten_path)) {
```
Comment (ol0lll) [inline entity_reference_override.module]: `$entity->overwritten_path` could be `$entity->entity_reference_override`

File: entity_reference_override.module [hunk 3]
```diff
@@ -35,3 +35,12 @@ function entity_reference_override_field_formatter_info_alter(array &$info) {
   $info['entity_reference_entity_id']['field_types'][] = 'entity_reference_override';
   $info['entity_reference_label']['field_types'][] = 'entity_reference_override';
 }
+
+/**
+ * Implements hook_entity_build_defaults_alter().
+ */
+function entity_reference_override_entity_build_defaults_alter(array &$build, EntityInterface $entity, $view_mode) {
+  if (isset($entity->overwritten_path)) {
+    $build['#cache']['keys'][] = 'overridden_reference_field:' . $entity->overwritten_path;
```
Comment (ol0lll) [inline entity_reference_override.module]: `$build['#cache']['keys'][] = 'entity_reference_override:' . $entity-> entity_reference_override;`

**Review by ol0lll — CHANGES_REQUESTED** [review]

**Review by ol0lll — APPROVED** [review]
lgtm

**Pull request merged**